redbot/core/utils/__init__.py

```diff
@@ -424,3 +426,67 @@ async def without_duplicates(self) -> AsyncIterator[_T]:
                 yield item
                 _temp.add(item)
         del _temp
+
+    async def find(
+        self, predicate: Union[Callable, Coroutine], default: Optional[Any] = None
```

Comment:
Type hint should specify what should the function `predicate` accept and return:
```suggestion
        self,
        predicate: Callable[[_T], Union[bool, Awaitable[bool]]],
        default: Optional[Any] = None,
```